core/controllers/topics_and_skills_dashboard.py

```diff
@@ -48,7 +48,9 @@ class TopicsAndSkillsDashboardPageDataHandler(base.BaseHandler):
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
-    @acl_decorators.can_access_topics_and_skills_dashboard
+    # As anyone should be able to see the topics and skills list when creating
```

Comment:
I'm reading this function more carefully. It doesn't look like it returns the skills list. It only returns untriaged/unpublished skills. It also returns a lot of data that is not relevant to what you need in the creator dashboard.

More generally, I think it's not a good idea to couple this handler with the creator dashboard. Could we instead just leave this data handler as-is, and create a new handler for the creator dashboard that returns specifically the info that is required? Otherwise it's not clear to developers working on this handler that the handler serves a completely different unrelated page as well.

Reply:
@nithusha21 have we addressed this?

Reply:
Done

Reply:
I moved this data retrieval to the creator dashboard data handler itself (don't know why I didn't think of that earlier). It fits into the pattern used to retrieve exploration, and collection data in the creator dashboard.